src/btree/bt_sync_obsolete.c

```diff
@@ -406,16 +406,19 @@ __checkpoint_cleanup_page_skip(
      * FIXME: Read internal pages from non-logged tables when the remove/truncate
      * operation is performed using no timestamp.
      */
+    if (addr.type == WT_ADDR_LEAF_NO)
+        *skipp = true;
+    else if (addr.ta.newest_stop_durable_ts == WT_TS_NONE)
+        /* Only process logged tables when checkpoint cleanup is configured to be aggressive. */
+        *skipp = !(F_ISSET(S2C(session), WT_CONN_CKPT_CLEANUP_RECLAIM_SPACE) &&
+          F_ISSET(S2BT(session), WT_BTREE_LOGGED));
```

Comment:
Sorry but I'm not following the correlation comment and the code.
This code reads to me:
skip = true if reclaim == false and logged == false
but the comment appears to indicate the opposite.

Reply:
Yes, if we are not operating in aggressive mode (reclaim == false), we want to skip.

Reply:
It looks like the condition is hard to read, I am happy to flip it and remove the `!` at the very start as suggested [here](https://github.com/wiredtiger/wiredtiger/pull/10918#discussion_r1693910372).

Reply:
Would you consider the following:
```
/* top of function */
const unsigned int AGGRESSIVE_MODE = (WT_CONN_CKPT_CLEANUP_RECLAIM_SPACE|WT_BTREE_LOGGED);

*skipp = (F_MASK(S2C(session), AGGRESSIVE_MODE) != AGGRESSIVE_MODE);
```

Reply:
I'd like to keep the notion of aggressive mode and logging separate. I can imagine that we may do different things unrelated to logged tables if the aggressive mode is enabled.

Reply:
Updated condition in 204385f31b5ff05a56e997fbfedf199a40c046a6